flepimop/gempyor_pkg/src/gempyor/resume_pull.py

```diff
@@ -0,0 +1,235 @@
+
+    Notes:
+    - Ensure that AWS credentials are configured properly for boto3 to access the S3 bucket.
+    - This function assumes that the S3 URIs provided are in the format `s3://bucket-name/path/to/object`.
+    """
+    s3 = boto3.client("s3")
+    for s3_uri in file_name_map:
+        bucket = s3_uri.split("/")[2]
+        object = s3_uri[len(bucket) + 6 :]
+        try:
+            s3.head_object(Bucket=bucket, Key=object)
+            if os.path.exists(file_name_map[s3_uri]) is False:
+                raise FileExistsError(f"For {s3_uri}, it is not copied to {file_name_map[s3_uri]}.")
+        except botocore.exceptions.ClientError as e:
+            if e.response["Error"]["Code"] == "404":
+                print(f"Input {s3_uri} does not exist.")
+            else:
+                raise
+
+
+# Todo: Unit Test
+def pull_check(file_name_map: Dict[str, str]) -> None:
+    """
+    Verifies the existence of specified source files and checks if corresponding destination files are present.
+    If a source file does not exist or the destination file is missing, it raises appropriate errors or prints a message.
+
+    Parameters:
+    file_name_map (Dict[str, str]): A dictionary where the keys are source file paths and the values are the corresponding destination file paths.
+
+    Dependencies:
+    - os: The standard library module for interacting with the operating system, used here to check for file existence.
+
+    Functionality:
+    1. Iterate through Source Files: For each source file path in the `file_name_map` dictionary:
+       - Check if Source File Exists: Uses `os.path.exists` to check if the source file exists.
+       - Check Destination File Existence: If the source file exists, it checks if the corresponding destination file exists using `os.path.exists`.
+       - Handle Errors:
+         - If the source file does not exist, it prints a message indicating the missing source file.
+         - If the destination file does not exist, it raises a `FileExistsError` indicating the destination file is missing.
+
+    Example Usage:
+    file_name_map = {
+        "/path/to/source1.txt": "/path/to/destination1.txt",
+        "/path/to/source2.txt": "/path/to/destination2.txt"
+    }
+
+    pull_check(file_name_map)
+
+    Exceptions:
+    - FileExistsError: Raised if the corresponding destination file for an existing source file is missing.
+
+    Notes:
+    - Ensure that the paths provided are valid and accessible on the file system.
+    """
+    for src_file in file_name_map:
+        if os.path.exists(src_file):
+            if os.path.exists(file_name_map[src_file]) is False:
+                raise FileExistsError(f"For {src_file}, it is not copied to {file_name_map[src_file]}.")
+        else:
+            print(f"Input {src_file} does not exist.")
+
+
+if __name__ == "__main__":
+    fetching_resume_files()
```

Comment:
@jcblemai  Will users call this command to pull resume file directly？ Should I make it callable？

Reply:
it's kind of conserved stylistic convention.

in my opinion, i think that's ...okay as a convention, but that we should have an overhaul that re-routes all of these kind of invocations through our click framework + issues a warning to users that an execution should happen that way.

Reply:
I don't want to belabor this too much because this is minor, but I am going to push back on this being "stylistic convention". These two lines are the difference between allowing users to interact with this script directly vs not:
```
twillard@Mac ~/D/G/H/f/f/gempyor_pkg (python_script_resume) [1]> python src/gempyor/resume_pull.py --help
Usage: resume_pull.py [OPTIONS]

Options:
  --resume_location TEXT     the path for the last run's output  [required]
  --discard_seeding BOOLEAN  required bool value for discarding seeding or not
                             [required]
  --block_index INTEGER      [required]
  --resume_run_index TEXT    [required]
  --flepi_run_index TEXT     [required]
  --flepi_prefix TEXT        [required]
  --help                     Show this message and exit.
twillard@Mac ~/D/G/H/f/f/gempyor_pkg (python_script_resume)>
```
vs
```
twillard@Mac ~/D/G/H/f/f/gempyor_pkg (python_script_resume)> python src/gempyor/resume_pull.py --help                    
twillard@Mac ~/D/G/H/f/f/gempyor_pkg (python_script_resume)>
```
If the goal is to have a consistent CLI then we shouldn't give users the option to run the same tool in multiple ways.

Reply:
fair, not exactly "style" - but the current library convention seems to be making many (most? almost all?) of the scripts have a main.

i am on-board with deleting all of those (agree - definitely want fewer interface points to maintain), but that is a breaking change and should probably be done as its own issue/PR.

Reply:
This main is removed.